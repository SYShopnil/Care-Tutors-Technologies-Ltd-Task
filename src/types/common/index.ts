export enum EDataTestId {
  SHeaderMain = "SHeaderMain",
  CToDoCreateUpdate = "CToDoCreateUpdate",
  CTodoItem = "CTodoItem",
  CInputForm = "CInputForm",
  CToDoList = "CToDoList",
  CButton = "CButton",
}
