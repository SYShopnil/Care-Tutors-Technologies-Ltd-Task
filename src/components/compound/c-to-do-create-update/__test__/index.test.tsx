import { render, screen } from "@testing-library/react";
import { CToDoCreateAndUpdate } from "..";
import { EDataTestId } from "@src/types/common";
import {
  EToDoMode,
  ICEditTodo,
} from "@src/types/compound/c-to-do-create-update";
import CStoreProvider from "../../c-store-provider";

const mock_update: ICEditTodo = {
  mode: EToDoMode.UPDATE,
  onClose: () => {
    console.log(`hello`);
  },
  todo: {
    description: "Hello this is description",
    id: "12345",
    title: "Hello this is a title",
  },
};
describe("When To Do Update and Create Component Rendered", () => {
  it("Expect render successfully", async () => {
    render(
      <CStoreProvider>
        <CToDoCreateAndUpdate {...mock_update} />
      </CStoreProvider>
    );
    const myElement = screen.getByRole(EDataTestId.CToDoCreateUpdate);
    expect(myElement).toMatchSnapshot();
  });
});
