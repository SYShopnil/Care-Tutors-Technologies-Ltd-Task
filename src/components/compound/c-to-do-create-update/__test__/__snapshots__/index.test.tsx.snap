// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`When To Do Update and Create Component Rendered Expect render successfully 1`] = `
<div
  role="CToDoCreateUpdate"
>
  <form
    class="space-y-4"
    role="CInputForm"
  >
    <input
      class="w-full p-2 border rounded"
      placeholder="Title"
      type="text"
      value="Hello this is a title"
    />
    <textarea
      class="w-full p-2 border rounded"
      placeholder="Description"
    >
      Hello this is description
    </textarea>
    <div
      class="flex justify-center items-center"
    >
      <span
        class="cursor-pointer"
        role="CButton"
      >
        <span
          class="flex justify-center items-center  "
        >
          <span
            class="text-base md:text-base_2  flex justify-center items-center px-[1rem] py-[0.5rem] rounded-[0.563rem] w-[100%] h-[100%] bg-[#EBF5EC] hover:bg-[#7F4D4F]  duration-[0.5s] text-[#7F4D4F] hover:text-[#EBF5EC]   border-[0.063rem] hover:border-[#EBF5EC] border-[#7F4D4F]   "
          >
            <span>
              Update
            </span>
          </span>
        </span>
      </span>
    </div>
  </form>
</div>
`;
